lib/straight-server/gateway.rb

```diff
@@ -21,32 +21,47 @@ def initialize_exchange_rate_adapters
         self.exchange_rate_adapter_names.each do |adapter|
           begin
             @exchange_rate_adapters << Straight::ExchangeRate.const_get("#{adapter}Adapter").instance
-          rescue NameError => e
-            puts "WARNING: No exchange rate adapter with the name #{adapter} was found!"
+          rescue NameError
+            StraightServer.logger.warn "No exchange rate adapter with the name #{adapter} was found!"
           end
         end
       end
     end
 
     def initialize_blockchain_adapters
-      @blockchain_adapters = []
-      StraightServer::Config.blockchain_adapters.each do |name, attrs|
-
-        adapter = Straight::Blockchain.const_get("#{name}Adapter")
-        next unless adapter
-
-        @blockchain_adapters <<
-          case name
-          when 'ChainCom'
-            adapter.mainnet_adapter(api_key_id: attrs['api_key_id'])
-          when 'Insight'
-            adapter.mainnet_adapter(main_url: attrs['mainnet_url'], test_url: attrs['testnet_url'])
-          else
-            adapter.mainnet_adapter
+      all_instances = {main: [], test: []}
```

Comment:
> Style guides also suggest hashrocket syntax

That was irony about things like https://github.com/styleguide/ruby. As you can see, they are subject to change.

> you work on open source for community

Wrong. I'm working on the product for end-users and I care about delivering features without bugs, not about opinions on where I should put spaces.